sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/NetworkInterfaceOperationsTests.java

```diff
@@ -509,6 +511,113 @@ public void canAssociateNatGateway() {
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingPrimaryNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrimaryPrivateIPAddressDynamic()
+            .withNewPrimaryPublicIPAddress()
+            .defineSecondaryIPConfiguration("secondary")
+            .withExistingNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrivateIpAddressDynamic()
+            .withNewPublicIpAddress()
+            .withPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
+            .attach()
+            .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
+            .create();
+
+        nic.refresh();
+        List<NetworkInterfaceIpConfigurationInner> ipConfigurationInnerList =  nic.innerModel().ipConfigurations()
+            .stream().filter(ipConfig -> DeleteOptions.DELETE.equals(ipConfig.publicIpAddress().deleteOption()))
+            .collect(Collectors.toList());
+
+        Assertions.assertEquals(2, ipConfigurationInnerList.size());
+    }
+
+    @Test
+    public void canUpdateNicWithMultipleDeleteOptions() {
+        String subnetName = generateRandomResourceName("subnet-", 15);
+        resourceManager.resourceGroups().define(rgName).withRegion(Region.US_EAST).create();
+        Network vnet = networkManager.networks()
+            .define(generateRandomResourceName("vnet-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withAddressSpace("10.0.0.0/28")
+            .withSubnet(subnetName, "10.0.0.0/28")
+            .create();
+
+        NetworkInterface nic = networkManager.networkInterfaces()
+            .define(generateRandomResourceName("nic-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingPrimaryNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrimaryPrivateIPAddressDynamic()
+            .withNewPrimaryPublicIPAddress()
+            .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
+            .defineSecondaryIPConfiguration("secondary1")
+                .withExistingNetwork(vnet)
+                .withSubnet(subnetName)
+                .withPrivateIpAddressDynamic()
+                .withNewPublicIpAddress()
+                .withPublicIPAddressDeleteOptions(DeleteOptions.DETACH)
+                .attach()
+            .defineSecondaryIPConfiguration("secondary2")
+                .withExistingNetwork(vnet)
+                .withSubnet(subnetName)
+                .withPrivateIpAddressDynamic()
+                .withNewPublicIpAddress()
+                .withPublicIPAddressDeleteOptions(DeleteOptions.DETACH)
+                .attach()
+            .create();
+
+        nic.refresh();
+        nic.update().withNewPrimaryPublicIPAddress().withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DETACH).apply();
```

Comment:
We could add an assertion here, that the new primary PublicIP is different than the original one, e.g. PublicIpAddress.id() are different.